guides/micronaut-data-mongodb/java/src/main/java/example/micronaut/domain/Student.java

```diff
@@ -0,0 +1,80 @@
+package example.micronaut.domain;
+
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.data.annotation.GeneratedValue;
+import io.micronaut.data.annotation.Id;
+import io.micronaut.data.annotation.MappedEntity;
+import io.micronaut.data.annotation.Relation;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+@MappedEntity("students")
+public class Student {
+
+    @Id
+    @GeneratedValue
+    private String id;
+
+    private String name;
+
+    @Relation(value = Relation.Kind.MANY_TO_MANY, cascade = Relation.Cascade.PERSIST)
+    private List<Course> courses;
+
+    public Student(String name, @Nullable List<Course> courses) {
+        this.name = name;
+        this.courses = courses == null ? new ArrayList<>() : courses;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public List<Course> getCourses() {
```

Comment:
i assume this can be null if the relation is not fetched. If yes, annotate it iwth `@Nullable`